hydra-main/src/main/java/com/addthis/hydra/job/store/CachedSpawnDataStore.java

```diff
@@ -0,0 +1,151 @@
+                        String path = key.getLeft();
+                        String childId = key.getRight();
+
+                        if (childId == null) {
+                            return CachedSpawnDataStore.this.dataStore.get(path);
+                        } else {
+                            return CachedSpawnDataStore.this.dataStore.getChild(path, childId);
+                        }
+                    }
+                });
+    }
+
+    @Override
+    public String getDescription() {
+        return dataStore.getDescription();
+    }
+
+    @Override
+    public String get(String path) {
+        try {
+            return cache.get(defaultKey(path));
+        } catch (ExecutionException | CacheLoader.InvalidCacheLoadException e) {
+            log.error("failed to execute get from cache", e);
+            return null;
+        }
+    }
+
+    @Override
+    public Map<String, String> get(String[] paths) {
+        List<String> notCached = new ArrayList<>();
+        Map<String, String> results = new TreeMap<>();
+
+        // Load every path from the cache that we can. For all the paths that do not exist in the cache, continue to
+        // do the batched query logic which should be implemented in the underlying DataStore's `get(String[])`
+        for (String path : paths) {
+            String result = cache.getIfPresent(defaultKey(path));
+            if (result == null) {
+                notCached.add(path);
+            } else {
+                results.put(path, result);
+            }
+        }
+
+        String[] remainingPaths = new String[notCached.size()];
+        remainingPaths = notCached.toArray(remainingPaths);
+
+        Map<String, String> resultsFromDB = dataStore.get(remainingPaths);
+
+        for (Map.Entry<String, String> entry : resultsFromDB.entrySet()) {
+            String path = entry.getKey();
+            cache.put(defaultKey(path), entry.getValue());
+        }
+
+        results.putAll(resultsFromDB);
+        return results;
+    }
+
+    @Override
+    public void put(String path, String value) throws Exception {
+        dataStore.put(path, value);
+        cache.put(defaultKey(path), value);
+    }
+
+    @Override
+    public void putAsChild(String parent, String childId, String value) throws Exception {
+        dataStore.putAsChild(parent, childId, value);
+        cache.put(ImmutablePair.of(parent, childId), value);
+    }
+
+    @Override
+    public String getChild(String parent, String childId) throws Exception {
+        return cache.get(ImmutablePair.of(parent, childId));
+    }
+
+    @Override
+    public void deleteChild(String parent, String childId) {
+        cache.invalidate(ImmutablePair.of(parent, childId));
+        dataStore.deleteChild(parent, childId);
+    }
+
+    @Override
+    public void delete(String path) {
+        cache.invalidate(defaultKey(path));
+        dataStore.delete(path);
+    }
+
```

Comment:
In this case, I agree.. Mostly for the same reasons I voted for data store before cache on the earlier note.